rdrf/rdrf/templates/registration/registration_login_details.html

```diff
@@ -1,7 +1,29 @@
 {% load i18n static %}
+{% load get_bootstrap_select_i18n_file %}
+{% get_current_language as CURRENT_LANGUAGE %}
+
+<link rel="stylesheet" href="{% static 'vendor/bootstrap-select-1.14.0-beta3/css/bootstrap-select.min.css' %}">
 
 <script type="text/javascript" src="{% static 'js/show_password.js' %}"></script>
+<script id="id_bootstrap_selector_js" type="text/javascript" src="{% static 'vendor/bootstrap-select-1.14.0-beta3/js/bootstrap-select.min.js' %}"></script>
+<script type="text/javascript" src="{{ CURRENT_LANGUAGE|get_bootstrap_select_i18n_file }}"></script>
+
 <script>
+    function constructPreferredLanguages() {
+        const browser_language = navigator.languages[0];
+        const not_in_languages_array = !{{ all_language_codes|safe }}.includes(browser_language);
+        const getPureLanguageCode = (language) => language.split('-')[0];
+
+        {{ preferred_languages|safe }}.forEach((language) => {
+            $("#id_preferred_languages").append(`
```

Comment:
`append` can be called once with an array of elements, rather than for each element.